### PR TITLE
[SDK] - allow passing Chain object to initialize SDK

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

File: packages/sdk/src/evm/core/sdk.ts [hunk 1]
```diff
@@ -159,10 +159,23 @@ export class ThirdwebSDK extends RPCConnectionHandler {
...
-    network: NetworkInput,
+    network: NetworkInput | Chain,
     options: SDKOptions = {},
     storage: ThirdwebStorage = new ThirdwebStorage(),
   ) {
+    if (
+      typeof network !== "string" &&
+      typeof network !== "number" &&
+      !ethers.Signer.isSigner(network) &&
+      !ethers.providers.Provider.isProvider(network)
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/sdk.ts:170]: Wonder if it's safer to check if object and has rpc property in there with at least one rpc?

File: packages/sdk/test/evm/sdk-init.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { Goerli, Fncy } from "@thirdweb-dev/chains";
+// import { signers, storage } from "./before-setup";
+// import { AddressZero } from "@ethersproject/constants";
+// import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+
+global.fetch = require("cross-fetch");
+
+describe("SDK Initialization", async () => {
+  // let adminWallet: SignerWithAddress, samWallet: SignerWithAddress;
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/sdk-init.ts:11]: Cleanup

Comment (jnsdls) [reply packages/sdk/test/evm/sdk-init.ts:11]: was gonna also test it with signer etc but yeah

File: packages/sdk/src/evm/core/sdk.ts (hunk 1 above)
Comment (jnsdls) [reply packages/sdk/src/evm/core/sdk.ts:170]: I'm scared that `Signer` JS typeof would also be `object` (same for provider)

**Pull request merged**